### PR TITLE
Np 46927, FetchNviInstitutionReportPresigner

### PR DESCRIPTION
**Pull request by monaullah**
Jira subtask: _NP-46927 Use ApiS3PresignerGatewayHandler in report-api_

Changes:
- `FetchNviInstitutionReportPresigner` generates presigned s3 url and sends sqs message to trigger generation og nvi institution report
- Added new s3 bucket where the file will be persisted: `nvi-reports`
- Renamed env var `AWS_REGION_NAME` tp `AWS_REGION` (same as in `nva-commons`)

In next PR: implement handling of sqs message and persisting report in s3 bucket in `FetchNviInstitutionReport`

### CODE REVIEW

File: report-api/src/main/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReport.java [hunk 1]
```diff
@@ -51,6 +52,12 @@ protected List<MediaType> listSupportedMediaTypes() {
         return List.of(TEXT_CSV, TEXT_PLAIN, MICROSOFT_EXCEL, OOXML_SHEET);
     }
 
+    @Override
+    protected void validateRequest(Void unused, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        //This handler is in lambda subnet in vpc, and only triggered via SQS via FetchDataReportPresigner
+        //TODO: Validate input
```
Comment (ketilaa) [inline report-api/src/main/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReport.java:58]: Remove the TODO if there is not input to validate?

File: report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                                             randomUri());
+        var output = new ByteArrayOutputStream();
+        var context = new FakeContext();
+        handler.handleRequest(request, output, context);
+        assertTrue(logAppender.getMessages().contains("reporting year: " + SOME_YEAR));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {TEXT_CSV, TEXT_PLAIN, OPEN_XML, EXCEL})
+    void shouldReturnPreSignedUrl(String contentType) throws IOException {
```
Comment (ketilaa) [inline report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java:96]: Test for non-supported content types?

**Review by ketilaa — APPROVED** [review]
Have a look at my comments and consider.

File: report-api/src/main/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportPresigner.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+
+public class FetchNviInstitutionReportPresigner extends ApiS3PresignerGatewayHandler<Void> {
+
+    private static final Logger logger = LoggerFactory.getLogger(FetchNviInstitutionReport.class);
+    private static final Duration SIGN_DURATION = Duration.ofMinutes(60);
```
Comment (brinxmat) [inline report-api/src/main/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportPresigner.java:28]: I wonder if this is not echoing a param in the SAM?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java (hunk 2 above)
Comment (monaullah) [reply report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java:96]: Do you mean wrong code `302` in commons @brinxmat?

Comment (ketilaa) [reply report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java:96]: Then all is good.

**Pull request merged**